src/atama/receiver/node/receiver_node.cpp

```diff
@@ -0,0 +1,113 @@
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+#include <memory>
+#include <set>
+#include <string>
+#include <vector>
+
+#include "atama/receiver/node/receiver_node.hpp"
+#include "kansei_interfaces/msg/orientation.hpp"
+#include "ninshiki_interfaces/msg/detected_object.hpp"
+#include "ninshiki_interfaces/msg/detected_objects.hpp"
+#include "rclcpp/rclcpp.hpp"
+#include "tachimawari_interfaces/srv/get_joints.hpp"
+
+using namespace std::chrono_literals;
+
+namespace atama::receiver
+{
+
+ReceiverNode::ReceiverNode(rclcpp::Node::SharedPtr node, std::shared_ptr<atama::head::Head> head)
+: node(node), head(head)
+{
+  using ninshiki_interfaces::msg::DetectedObject;
+
+  get_joints_client = node->create_client<GetJoints>("/joint/get_joints");
+
+  get_orientation_subsciber = node->create_subscription<Orientation>(
+    "measurement/orientation", 10,
+    [this](const Orientation::SharedPtr message) {
+      // this->head->set_yaw(message->orientation[2]);
+    }
+  );
+
+  get_detection_result_subsciber =
+    node->create_subscription<DetectedObjects>(
+    "ninshiki_py/detection", 10,
+    [this](const DetectedObjects::SharedPtr message) {
+      std::vector<DetectedObject> temp_detection_result;
+
+      for (const auto & detected_object : message->detected_objects) {
+        temp_detection_result.push_back(detected_object);
+      }
+
+      this->head->detection_result = temp_detection_result;
+    }
+    );
+}
+
+bool ReceiverNode::get_joints_data()
+{
+  using tachimawari::joint::Joint;
+  using tachimawari::joint::JointId;
+
+  while (!get_joints_client->wait_for_service(1s)) {
+    if (rclcpp::ok()) {
+      // service not available, waiting again...
+    } else {
+      // Interrupted while waiting for the service. Exiting.
+      return false;
+    }
+  }
+  auto result = get_joints_client->async_send_request(
+    std::make_shared<tachimawari_interfaces::srv::GetJoints::Request>());
+
+  if (rclcpp::spin_until_future_complete(node, result) ==
+    rclcpp::FutureReturnCode::SUCCESS)
+  {
+    std::vector<Joint> temp_joints;
+    std::set<uint8_t> joints_id;
+
+    for (std::string id : {"neck_yaw", "neck_pitch"}) {
+      joints_id.insert(JointId::by_name.find(id)->second);
+    }
+
+    for (const auto & joint : result.get()->joints) {
+      // Joint Id found in joints_id set
+      if (joints_id.find(joint.id) != joints_id.end()) {
+        temp_joints.push_back(Joint(joint.id, joint.position));
+      }
+    }
+
+    head->set_joints(temp_joints);
+    return true;
+  } else {
+    // Failed to call service
+    return false;
+  }
+}
+
+std::string ReceiverNode::get_node_prefix() const
+{
+  return "receiver";
+}
```

Comment:
i suggest to use static const property rather than a non static function here

see this too:
https://github.com/ichiro-its/atama/blob/daeb9e8c86895e2dd3cef4876b8cb93199828a43/src/atama/sender/node/sender_node.cpp#L159-L162

Reply:
Please check again, I have already changed it but I don't know if that is what you mean or not.

Reply:
> Please check again, I have already changed it but I don't know if that is what you mean or not.

use this one, inside the header declaration
```c++
static std::string get_node_prefix() const;
```

Reply:
It raised an error when I write `static` function with `const`. From what I search on the internet, it can't write in that format. Is there any suggestion?

Reply:
i forgot about that, let's just make it `static` without `const`